api/http/handler/kubernetes/volumes.go

```diff
@@ -0,0 +1,136 @@
+package kubernetes
+
+import (
+	"fmt"
+	"net/http"
+
+	models "github.com/portainer/portainer/api/http/models/kubernetes"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+
+	httperror "github.com/portainer/portainer/pkg/libhttp/error"
+	"github.com/portainer/portainer/pkg/libhttp/request"
+	"github.com/portainer/portainer/pkg/libhttp/response"
+)
+
+// @id GetAllKubernetesVolumes
+// @summary Get Kubernetes volumes within the given Portainer environment
+// @description Get a list of all kubernetes volumes within the given environment (Endpoint). The Endpoint ID must be a valid Portainer environment identifier.
+// @description **Access policy**: Authenticated user.
+// @tags kubernetes
+// @security ApiKeyAuth || jwt
+// @produce json
+// @param id path int true "Environment identifier"
+// @param withApplications query boolean false "When set to True, include the applications that are using the volumes. It is set to false by default"
+// @success 200 {object} map[string]portainer.K8sVolumeInfo "Success"
+// @failure 400 "Invalid request payload, such as missing required fields or fields not meeting validation criteria."
+// @failure 403 "Unauthorized access or operation not allowed."
+// @failure 500 "Server error occurred while attempting to retrieve kubernetes volumes."
+// @router /kubernetes/{id}/volumes [get]
+func (handler *Handler) GetAllKubernetesVolumes(w http.ResponseWriter, r *http.Request) *httperror.HandlerError {
+	volumes, err := handler.getKubernetesVolumes(r)
+	if err != nil {
+		return err
+	}
+
+	return response.JSON(w, volumes)
+}
+
+// @id getAllKubernetesVolumesCount
+// @summary Get the total number of kubernetes volumes within the given Portainer environment.
+// @description Get the total number of kubernetes volumes within the given environment (Endpoint). The total count depends on the user's role and permissions. The Endpoint ID must be a valid Portainer environment identifier.
+// @description **Access policy**: Authenticated user.
+// @tags kubernetes
+// @security ApiKeyAuth || jwt
+// @produce json
+// @param id path int true "Environment identifier"
+// @success 200 {integer} integer "Success"
+// @failure 400 "Invalid request payload, such as missing required fields or fields not meeting validation criteria."
+// @failure 403 "Unauthorized access or operation not allowed."
+// @failure 500 "Server error occurred while attempting to retrieve kubernetes volumes count."
+// @router /kubernetes/{id}/volumes/count [get]
+func (handler *Handler) getAllKubernetesVolumesCount(w http.ResponseWriter, r *http.Request) *httperror.HandlerError {
+	volumes, err := handler.getKubernetesVolumes(r)
```

Comment:
I wont fix yet, but it would be worth being able to pass in options for the cli get volumes function. The dashboard api was able to get the count of all items (e.g. pods), without needing the full list returned - so the payload was much smaller and ran faster.